lib/schedule/service/isolate_schedule_source_decorator.dart

```diff
@@ -42,9 +45,10 @@ class IsolateScheduleSourceDecorator extends ScheduleSource {
     final subscription = _isolateToMain!.listen((result) {
       cancellationToken.setCancellationCallback(null);
 
-      if (result != null && result is! ScheduleQueryResult) {
+      // TODO: [Leptopoda] validate changes
+      if (result == null || result is! ScheduleQueryResult) {
         potentialException = ScheduleQueryFailedException(result);
-        completer.complete(null);
+        completer.complete();
       } else {
         completer.complete(result);
       }
```

Comment:
I only changed the logic here but it didn't really make sense the other way around.
Please verify that this is still the wanted implementation.